src/domUpdates.js

```diff
@@ -42,9 +45,66 @@ let domUpdates = {
     })
   },
 
-  displayShoppingList(ingredientsNeeded) {
-    makeRecipeMessage.innerText = `You do not have enough ingredients in your pantry to make this recipe.`
-    // shopping list functionality shows up here... add button so you can click to expand?
+  displayShoppingList(shoppingList) {
+    modalShoppingMessageYes.style.display = 'none'
+    modalShoppingList.style.display = 'block'
+    shoppingList.forEach(shoppingItem => {
+      const listItem = document.createElement('tr')
+      const itemName = document.createElement('td')
+      const itemQuantity = document.createElement('td')
+      const itemCostPerUnit = document.createElement('td')
+      const itemTotalCost = document.createElement('td')
+      let unit = this.formatUnits(shoppingItem.unit)
+      let totalCost = (shoppingItem.cost * shoppingItem.quantity)/100
+
+      itemName.innerText = shoppingItem.name
+      itemQuantity.innerText = `${shoppingItem.quantity} ${unit}`
+      itemCostPerUnit.innerText = `$${shoppingItem.cost/100}`
+      itemTotalCost.innerText = `$${totalCost.toFixed(2)}`
+
+      itemCostPerUnit.classList.add("price")
+      itemTotalCost.classList.add("price")
+
+      modalShoppingItems.appendChild(listItem)
+      listItem.appendChild(itemName)
+      listItem.appendChild(itemQuantity)
+      listItem.appendChild(itemCostPerUnit)
+      listItem.appendChild(itemTotalCost)
+    })
+
+    this.displayTotalCost(shoppingList)
+  },
+
+  displayTotalCost(shoppingList) {
+    const totalListCost = shoppingList.reduce((totalCost, item) => {
+      totalCost += (item.cost * item.quantity) / 100
+      return totalCost
+    }, 0)
+
+    modalTotalCost.innerHTML = `
```

Comment:
I don't know if the use of innerHTML is frowned upon generally (I was under the impression it was, but i could be wrong) so I'd say we could have this section be similar to lines 52, 53, 68, and 69 but it's up to the team if we want to do that or leave it as innerHTML.

Reply:
I've never heard of it being frowned upon! Have used it in several past projects now and instructors have never said anything. But happy to adjust if you've heard otherwise.